test/access/manager/AccessManager.test.js

```diff
@@ -0,0 +1,523 @@
+        expect(groups).to.not.include(group);
+      });
+
+      it(`user cannot renounce other user's groups`, async function () {
+        await expectRevert(
+          this.manager.renounceGroup(user1, group, { from: user2 }),
+          'can only renounce roles for self',
+        );
+        await expectRevert(
+          this.manager.renounceGroup(user2, group, { from: user1 }),
+          'can only renounce roles for self',
+        );
+      });
+
+      it('cannot revoke public group', async function () {
+        await expectRevert(
+          this.manager.revokeGroup(user1, PUBLIC_GROUP, { from: admin }),
+          'AccessManager: irrevocable group',
+        );
+      });
+
+      it('cannot revoke nonexistent group', async function () {
+        await expectRevert(
+          this.manager.revokeGroup(user1, otherGroup, { from: admin }),
+          'AccessManager: unknown group',
+        );
+        await expectRevert(
+          this.manager.renounceGroup(user1, otherGroup, { from: user1 }),
+          'AccessManager: unknown group',
+        );
+      });
+    });
+
+    describe('querying', function () {
+      it('returns expected groups', async function () {
+        const getGroups = () => this.manager.getUserGroups(user1);
+
+        // only public group initially
+        expect(await getGroups()).to.equal('0x8000000000000000000000000000000000000000000000000000000000000000');
+
+        await this.manager.createGroup('0', '0', { from: admin });
+        await this.manager.grantGroup(user1, '0', { from: admin });
+        expect(await getGroups()).to.equal('0x8000000000000000000000000000000000000000000000000000000000000001');
+
+        await this.manager.createGroup('1', '1', { from: admin });
+        await this.manager.grantGroup(user1, '1', { from: admin });
+        expect(await getGroups()).to.equal('0x8000000000000000000000000000000000000000000000000000000000000003');
+
+        await this.manager.createGroup('16', '16', { from: admin });
+        await this.manager.grantGroup(user1, '16', { from: admin });
+        expect(await getGroups()).to.equal('0x8000000000000000000000000000000000000000000000000000000000010003');
+      });
+    });
+  });
+
+  describe('allowing', function () {
+    const group = '1';
+    const groupMember = user1;
+    const selector = web3.eth.abi.encodeFunctionSignature('restrictedFunction()');
+    const otherSelector = web3.eth.abi.encodeFunctionSignature('otherRestrictedFunction()');
+
+    beforeEach('deploying managed contract', async function () {
+      await this.manager.createGroup(group, '', { from: admin });
+      await this.manager.grantGroup(groupMember, group, { from: admin });
+      this.managed = await AccessManaged.new(this.manager.address);
+    });
+
+    it('non-admin cannot change allowed groups', async function () {
+      await expectRevert(
+        this.manager.methods['setFunctionAllowedGroup(address,bytes4[],uint8,bool)'](
```

Comment:
Can we put all `setFunctionAllowedGroup(address,bytes4[],uint8,bool)` in a variable?